src/analysis/inner_product_space/basic.lean

```diff
@@ -145,12 +152,11 @@ include c
 local notation `⟪`x`, `y`⟫` := @inner 𝕜 F _ x y
 local notation `norm_sqK` := @is_R_or_C.norm_sq 𝕜 _
 local notation `reK` := @is_R_or_C.re 𝕜 _
-local notation `absK` := @is_R_or_C.abs 𝕜 _
 local notation `ext_iff` := @is_R_or_C.ext_iff 𝕜 _
 local postfix `†`:90 := star_ring_end _
 
 /-- Inner product defined by the `inner_product_space.core` structure. -/
-def to_has_inner : has_inner 𝕜 F := { inner := c.inner }
+def to_has_inner : has_inner 𝕜 F := c.to_has_inner
```

Comment:
Why not delete this and use the base-class projection?

Reply:
For some reason, both `inner_product_space` and `inner_product_space.core` are defined as `@[class] structure`, so the projection takes `c` as an explicit argument.